core/src/main/scala/org/apache/spark/status/AppStatusListener.scala

```diff
@@ -868,13 +868,17 @@ private[spark] class AppStatusListener(
     // check if there is a new peak value for any of the executor level memory metrics
     // for the live UI. SparkListenerExecutorMetricsUpdate events are only processed
     // for the live UI.
-    event.executorUpdates.foreach { case (_, peakUpdates) =>
+    event.executorUpdates.foreach { case (key, peakUpdates) =>
       liveExecutors.get(event.execId).foreach { exec =>
         if (exec.peakExecutorMetrics.compareAndUpdatePeakValues(peakUpdates)) {
           maybeUpdate(exec, now)
```

Comment:
Actually I wonder why here use `maybeUpdate` instead of `update`..

Reply:
You are right. Changed it to `update`. Thanks.

Reply:
I am not so familiar with `AppStatusListener`, so please educate me. From what I see, for most events, `maybeUpdate` is called. I assume that we only want to update a live application if it hasn't been updated in the last configured period. For the  `SparkListenerStageExecutorMetrics` event, according to Edwina's comment, this is only processed when reading event logs, so `update` is called in that case.

Reply:
We need `update` here since we know that new peak values are received (by checking `compareAndUpdatePeakValues`). If we use `maybeUpdate`, we may miss recording these new peak values.

Reply:
Thanks for the explanation.